### PR TITLE
54314 - create stubs for pos

### PR DESCRIPTION
**Pull request by Vishal14583**
- Added fleet manager stub 
- Added ESS stub
- Added FSS stub

### CODE REVIEW

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Helpers/EnumHelper.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System.ComponentModel;
+using System.Reflection;
+
+namespace UKHO.FmEssFssMock.API.Helpers
+{
+    public static class EnumHelper
+    {
+        public static string GetEnumDescription(Enum enumVal)
+        {
+            System.Reflection.MemberInfo[] memInfo = enumVal.GetType().GetMember(enumVal.ToString());
```
Comment (harshal11869) [inline UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Helpers/EnumHelper.cs:10]: Could we please give readable variable name

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Models/Request/AddFileRequest.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿
+namespace UKHO.PeriodicOutputService.Common.Models.Request
+{
+    public class AddFileRequest
+    {
+        public IEnumerable<KeyValuePair<String, string>> Attributes { get; set; }
```
Comment (harshal11869) [inline UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Models/Request/AddFileRequest.cs:6]:
```suggestion
        public IEnumerable<KeyValuePair<string, string>> Attributes { get; set; }
```

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Filters/BinaryRequestBodyFormatter.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+        public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputFormatterContext context)
+        {
+            var request = context.HttpContext.Request;
+            var contentType = context.HttpContext.Request.ContentType;
+
+            if (contentType == "application/octet-stream")
+            {
+                using var ms = new MemoryStream();
```
Comment (harshal11869) [inline UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Filters/BinaryRequestBodyFormatter.cs:31]:
```suggestion
                using var memoryStream = new MemoryStream();
```

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/TestConfiguration.cs [hunk 4]
```diff
@@ -6,8 +6,8 @@ public class TestConfiguration
     {
         protected IConfigurationRoot ConfigurationRoot;
         public FleetManagerB2BApiConfiguration fleetManagerB2BConfig = new();
-        public EssAuthorizationConfiguration EssAuthorizationConfig = new();
-        public FunctionalTestFSSApiConfiguration FssConfig = new();
+        public ESSApiConfiguration EssAuthorizationConfig = new();
+        public FSSApiConfiguration FssConfig = new();
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/TestConfiguration.cs:10]: This need to be corrected wherever it referenced

```suggestion
        public ESSApiConfiguration EssApiConfiguration = new();
        public FSSApiConfiguration FssApiConfiguration = new();
```

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs [hunk 5]
```diff
@@ -77,10 +78,10 @@ public async Task WhenICallTheExchangeSetApiWithInValidProductIdentifiers_ThenVa
         [Test]
         public async Task WhenICallTheFSSApiWithValidBatchId_ThenALargeMediaStructureIsCreated()
         {
-            HttpResponseMessage essApiResponse = await getproductIdentifier.GetProductIdentifiersDataAsync(ESSAuth.EssApiUrl, productIdentifiers, EssJwtToken);
+            HttpResponseMessage essApiResponse = await getproductIdentifier.GetProductIdentifiersDataAsync(ESSAuth.BaseUrl, productIdentifiers, EssJwtToken);
             Assert.That((int)essApiResponse.StatusCode, Is.EqualTo(200), $"Incorrect status code is returned {unpResponse.StatusCode}, instead of the expected status 200.");
 
-            DownloadedFolderPath = await FileContentHelper.CreateExchangeSetFileForLargeMedia(essApiResponse, FssJwtToken);
+            DownloadedFolderPath = await FileContentHelper.CreateExchangeSetFileForLargeMedia(essApiResponse, FssJwtToken,ESSBatchId);
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs:84]:
```suggestion
            DownloadedFolderPath = await FileContentHelper.CreateExchangeSetFileForLargeMedia(essApiResponse, FssJwtToken, ESSBatchId);
```

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs [hunk 6]
```diff
@@ -17,13 +17,14 @@ public class CallEssEndPoint
         private string FssJwtToken { get; set; }
         private GetProductIdentifiers getproductIdentifier { get; set; }
 
-        private static readonly EssAuthorizationConfiguration ESSAuth = new TestConfiguration().EssAuthorizationConfig;
-        private static readonly FunctionalTestFSSApiConfiguration FSSAuth = new TestConfiguration().FssConfig;
+        private static readonly ESSApiConfiguration ESSAuth = new TestConfiguration().EssAuthorizationConfig;
+        private static readonly FSSApiConfiguration FSSAuth = new TestConfiguration().FssConfig;
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs:21]: Is this ESS/FSS API Call Configuration or Authentication Configuration ?

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/TestConfiguration.cs (hunk 4 above)
Comment (Arit14975) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/Helpers/TestConfiguration.cs]: Updated.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs (hunk 5 above)
Comment (Arit14975) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs]: Updated.

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs (hunk 6 above)
Comment (Arit14975) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.API.FunctionalTests/FunctionalTests/CallEssEndPoint.cs]: Updated.

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Helpers/EnumHelper.cs (hunk 1 above)
Comment (Vishal14583) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Helpers/EnumHelper.cs]: Updated

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Models/Request/AddFileRequest.cs (hunk 2 above)
Comment (Vishal14583) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Models/Request/AddFileRequest.cs]: Corrected

File: UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Filters/BinaryRequestBodyFormatter.cs (hunk 3 above)
Comment (Vishal14583) [reply UKHO.FmEssFssMock.API/UKHO.FmEssFssMock.API/Filters/BinaryRequestBodyFormatter.cs]: Updated

**Review by mattwin1 — APPROVED** [review]
Reviewed pipeline-related files: Looks good. Approved.

**Review by martyn-fewtrell — APPROVED** [review]
Looks good

**Review by harshal11869 — APPROVED** [review]
LGTM

**Pull request merged**